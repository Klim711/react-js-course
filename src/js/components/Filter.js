import React, {Component} from 'react';

export default class Filter extends Component {
  constructor() {
    super();

    this.setActive = this.setActive.bind(this);
  }
  setActive(event) {
    this.props.setActive(event.target.value);
  }
  render () {
    const items = this.props.items.map((item) => {
      let className = (item === this.props.value) ? 'active' : '';

      return (
        <button className={className}
                key={item}
                value={item}
                onClick={this.setActive}>
          {item}
        </button>
      );
    });

    return (
        <div className="search__filter">
          <span className="search__filter__title">{this.props.content}</span>
          {items}
        </div>
    );
  }
}