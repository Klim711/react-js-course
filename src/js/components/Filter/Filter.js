import React, {Component} from 'react';

export default class Filter extends Component {
  constructor() {
    super();

    this.filterBy = this.filterBy.bind(this);
  }
  filterBy(event) {
    this.props.setSearchFilter(event.target.value);
  }
  render () {
    return (
        <div className="search__filter">
          Search By
          <button value="title" onClick={this.filterBy}>Title</button>
          <button value="genre" onClick={this.filterBy}>Genre</button>
        </div>
    );
  }
}