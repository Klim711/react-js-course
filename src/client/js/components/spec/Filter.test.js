import React from 'react';
import renderer from 'react-test-renderer';

import Filter from '../Filter';

describe('Filter component', () => {
  it('matches to snapshot', () => {
    const value = 'activeValue';
    const items = [
      'mockValue1',
      value,
      'mockValue2',
    ];

    const renderedValue = renderer.create(<Filter value={value} items={items}/>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
