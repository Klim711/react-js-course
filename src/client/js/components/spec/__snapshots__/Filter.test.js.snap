// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component matches to snapshot 1`] = `
<div
  className="search__filter"
>
  <span
    className="search__filter__title"
  />
  <button
    className=""
    onClick={[Function]}
    value="mockValue1"
  >
    mockValue1
  </button>
  <button
    className="active"
    onClick={[Function]}
    value="activeValue"
  >
    activeValue
  </button>
  <button
    className=""
    onClick={[Function]}
    value="mockValue2"
  >
    mockValue2
  </button>
</div>
`;
